config/src/generator.rs

```diff
@@ -52,3 +48,22 @@ pub fn validator_swarm_for_testing(nodes: usize) -> ValidatorSwarm {
     config.test = Some(TestConfig::open_module());
     validator_swarm(&NodeConfig::default(), nodes, [1u8; 32], true)
 }
+
+fn build_seed_peers(config: &NodeConfig) -> SeedPeersConfig {
+    let seed_config = config.validator_network.as_ref().unwrap();
+    let seed_handshake = 0;
+    let seed_pubkey = seed_config
+        .network_keypairs
+        .as_ref()
+        .unwrap()
+        .identity_keypair
+        .public_key();
+    let seed_base_addr = seed_config.advertised_address.clone();
+    let seed_addr = seed_base_addr.append_prod_protos(seed_pubkey, seed_handshake);
+
+    let mut seed_peers = SeedPeersConfig::default();
+    seed_peers
+        .seed_peers
+        .insert(seed_config.peer_id, vec![seed_addr]);
+    seed_peers
+}
```

Comment:
there's 3 build_seed_peers functions pretty much doing the same thing, maybe worth it to consolidate?

Reply:
+1 on this, not sure why we need three mostly identical functions

Reply:
... that's what i did previously, putting it in `NetworkConfig`

Reply:
blame @davidiw who has different taste of code!

Reply:
I think my feedback wasn't super clear :).

We don't need this for generator as mentioned above and the code in fullnode and validator swarm can be unified in config-builder. That would make it not so accessible, which is ideal since it has no practical use for non-test scenarios.